### DISCUSSION TITLE
Adding numbers to exercises in a series

### DISCUSSION THREAD

**Discussion by IsaacMe** [dodona-edu/feedback-and-support#9] [Feature requests]
At Hasselt University we use Dodona (and really like it) in addition to a studyguide that the students follow to complete the course. In our studyguide we refer to the exercises by using a number we appended in the title of each exercise on Dodona. However, this is not practical as the excercises are resued across several courses and the numbers do not always match.

We think it would be interesting to provide an option in Dodona to automatically add a numbering to exercises within a series/course. This will probably not only be useful to us, but also to other instructors who seek for short methods to easily refer to exercises from other sources.

Any thoughts on this suggestion?

**Comment by bmesuere** [comment]
@IsaacMe We have discussed this and have come up with the following proposal. Would this work for you?

**Change**
Add a setting to each series to enable the automatic numbering of exercises.

**Result**
Where possible, we add automatic sequential numbers to the UI:
- In the series listing on the course page
- In the exercise title when opening an exercise
- In the right hand side quick navigation menu

**Limitations**
- We don't allow manual numbers, only automatic ones
- This is optional and off by default
- We would number all items in a series. This might cause confusion in series containing both exercises and reading activities. For example, in a series with a reading activity, exercise, reading activity and exercise, "Solve the second exercise" might mean solve the exercise numbered as 2 (i.e. the first exercise) or the actual second exercise being the one numbered as 4.

**Reply by IsaacMe** [reply]
This sounds great and would certainly work for us. I think the limitations given do all make sense and would not pose any issues for us. So this addition would be perfect.

Thank you for looking into this, we really appreciate it.

**Reply by bmesuere** [reply]
👍 I have created an issue to track the implementation: #3431

**Comment by pdawyndt** [comment]
@bmesuere would you only number items in series, or also series in course?

**Reply by bmesuere** [reply]
Only exercises/reading activities. There is less need for series since you can name them yourself (not possible with exercises used in multiple courses) + the order depends on how you use the course (top-bottom or reverse).

**Comment by bmesuere** [comment]
This has since been implemented!
